### PR TITLE
[IoT Central] New API for 2022-06-30-preview release

### LINKED ISSUE #19403: [IoT Central] REST API Review (2022-06-30-preview)
Service name
IoT Central

Key contact for the review
[hayang@microsoft.com](mailto:hayang@microsoft.com), [Avneet.Singh@microsoft.com](mailto:Avneet.Singh@microsoft.com),[vincenzo.caruso@microsoft.com](mailto:vincenzo.caruso@microsoft.com)

Whether this is a new or existing API
Existing API

Whether or not your service already been deployed
No, TBD by end of June

A brief description of the material to be reviewed/changed:

1. Add Enrollment Group support
2. Add Scheduled Job Definition support
3. Add Dashboard support
4. Add Unmodeled Device Property support
5. Add Unmodeled Device Command support
6. Update existed Job add optional progress support.
7. Add new route for Device Group List Devices

[Swagger PR](https://github.com/Azure/azure-rest-api-specs/pull/19396)

### PR DESCRIPTION
**Pull request by yang-hai-feng**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>
Changes:
- Add new Enrollment Group support
- Add new Scheduled Job support
- Add Job progress support
- Add new Dashboard support
- Add unmodeled device twin/command support

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by yang-hai-feng** [comment]
> * EnrollmentGroups_Update

Thanks @mikekistler, my mistake put the response definition as Attestation instead of EnrollmentGroup, thanks for helping get the root cause!

File: specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2022-06-30-preview/iotcentral.json [hunk 1]
```diff
@@ -0,0 +1,8857 @@
...
+        "queryRange": {
+          "description": "The query range configuration of the chart",
+          "$ref": "#/definitions/TimeQueryRangeConfiguration"
+        }
+      },
+      "required": [
+        "queryRange"
+      ]
+    },
+    "BlobStorageV1Destination": {
```
Comment (markweitzel) [inline specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2022-06-30-preview/iotcentral.json:146]: @yang-hai-feng -- qq for you on something i just noticed... Why does the reference to `BlobStorageV1Destination` have the `V1` in it? This seems like it's exposing an implementation detail that can be hidden from the developer. Similar question for `DataExplorerV1Destination`

**Review by markweitzel — APPROVED** [review]
This looks good to me for preview.

File: specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2022-06-30-preview/iotcentral.json (hunk 1 above)
Comment (yang-hai-feng) [reply specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2022-06-30-preview/iotcentral.json:146]: Hi @markweitzel, this is currently designed for versioning the same type of destination if there is break changes in future, for example if we had new DataExplorerV2Destination defined with required broken shape, it requires to work together with the existed DataExplorerV1Destination, all contracts are exposed to end user directly to ensure their client code continue to work on v1 destination and also be able to create the v2 one. Also, we are holding this part in preview as now we are working on re-purpose the data export to a more generic concept of rule engine platform, the new destination could be an email, icm, flow trigger etc. We are preparing to join another API review office hours for this part changes when it is ready. Besides, we also got comments from 1.1-preview that our naming of entity (Data export related) is too long in this part, will be reconsider at same time of rule engine alignment.

**Review by JeffreyRichter — APPROVED** [review]

**Pull request merged**